components/ILIAS/Chatroom/classes/class.BannedUsersTable.php

```diff
@@ -0,0 +1,212 @@
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Data;
+use ILIAS\UI;
+use Psr\Http\Message\ServerRequestInterface;
+
+class BannedUsersTable
+{
+    private array $banned_users = [];
+    protected \ilCtrl $ctrl;
+    protected \ilLanguage $lng;
+    protected UI\Factory $ui_factory;
+    protected ServerRequestInterface $request;
+    protected Data\Factory $data_factory;
+
+    public function __construct(array $banned_users = [])
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->lng = $DIC->language();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->request = $DIC->http()->request();
+        $this->data_factory = new Data\Factory();
+        $this->banned_users = $banned_users;
+    }
+
+    public function getComponent(): UI\Component\Table\Data
+    {
+        $columns = $this->getColumns();
+        $actions = $this->getActions();
+        $data_retrieval = $this->getDataRetrieval();
+
+        $table = $this->ui_factory->table()
+                                  ->data($this->lng->txt('ban_table_title'), $columns, $data_retrieval)
+                                  ->withActions($actions)
+                                  ->withRequest($this->request);
+
+        return $table;
+    }
+
+    protected function getColumns(): array
+    {
+        return [
+            'user_id' => $this->ui_factory->table()->column()->number('User ID')
+                                          ->withIsSortable(true),
+
+            'login' => $this->ui_factory->table()->column()->text($this->lng->txt('login'))
+                                        ->withIsSortable(true),
+
+            'firstname' => $this->ui_factory->table()->column()->text($this->lng->txt('firstname'))
+                                            ->withIsSortable(true),
+
+            'lastname' => $this->ui_factory->table()->column()->text($this->lng->txt('lastname'))
+                                           ->withIsSortable(true),
+
+            'timestamp' => $this->ui_factory->table()->column()->text($this->lng->txt('chtr_ban_ts_tbl_head'))
+                                            ->withIsSortable(true),
+
+            'actor' => $this->ui_factory->table()->column()->text($this->lng->txt('chtr_ban_actor_tbl_head'))
+                                        ->withIsSortable(true),
+        ];
+    }
+
+    protected function getActions(): array
+    {
+        $query_params_namespace = ['chat_ban_table'];
+
+        $uri_detach = $this->data_factory->uri(
```

Comment:
Please change the implementation to the one I used for `SAML`. There should be an action handling method with a switch/match statement to delegate the action to the corresponding method (even if it is only `ban-delete` here) in the parent class (here: ilObjChatroomgui::class , please use the correct case here as well, even if PHP does not care).

Reply:
Done. That's true.